src/workerd/io/trace.c++

```diff
@@ -1481,17 +1481,21 @@ void SpanBuilder::setOperationName(kj::ConstString operationName) {
 
 void SpanBuilder::setTag(kj::ConstString key, TagValue value) {
   KJ_IF_SOME(s, span) {
-    auto keyPtr = key.asPtr();
-    s.tags.upsert(
-        kj::mv(key), kj::mv(value), [keyPtr](TagValue& existingValue, TagValue&& newValue) {
-      // This is a programming error, but not a serious one. We could alternatively just emit
-      // duplicate tags and leave the Jaeger UI in charge of warning about them.
-      [[maybe_unused]] static auto logged = [keyPtr]() {
-        KJ_LOG(WARNING, "overwriting previous tag", keyPtr);
-        return true;
-      }();
-      existingValue = kj::mv(newValue);
-    });
+    if (s.tags.size() >= Span::MAX_LOGS) {
```

Comment:
Note that this is technically a breaking change, but in practice we don't create more than 1023 tags internally. This is intended mostly as defensive programming in case we allow users to add their own span tags in the future. If this proves controversial for some reason we can drop it.